csmith-fuzzing/driver.py

```diff
@@ -1,33 +1,94 @@
-from subprocess import run, DEVNULL, PIPE
+import os, sys
+from subprocess import run, SubprocessError, DEVNULL, PIPE
+from tempfile import NamedTemporaryFile
 
 csmith_command = [
-        "csmith",
-        "--no-checksum",
-        "--nomain",
-        "--max-block-size", "1",
-        "--max-block-depth", "1",
-        "--output", "generated.h"]
+    "csmith",
+    "--no-checksum",
+    "--nomain",
+    "--max-block-size", "1",
+    "--max-block-depth", "1",
+]
 
-bindgen_command = ["bindgen", "generated.h"]
+def cat(path, title=None):
+    if not title:
+        title = path
+    print("-------------------- {} --------------------".format(title))
+    run(["cat", path])
 
-if __name__ == "__main__":
-    print("Bindgen fuzzing with csmith.")
-    print(
-        "This script will write to generated.h, bindgen_stdout, bindgen_stderr and platform.info . "
-        "These files can be deleted after running.")
+def run_logged(cmd):
+    with NamedTemporaryFile() as stdout, NamedTemporaryFile() as stderr:
+        result = run(cmd, stdin=DEVNULL, stdout=stdout, stderr=stderr)
+        if result.returncode != 0:
+            print()
+            print("Error: '{}' exited with code {}".format(" ".join(cmd), result.returncode))
+            cat(stdout.name, title="stdout")
+            cat(stdout.name, title="stderr")
+        return result
+
+def run_bindgen(input, output):
+    return run_logged([
+        "bindgen",
+        "--with-derive-partialeq",
+        "--with-derive-eq",
```

Comment:
Maybe we should use all of the `--with-derive` to improve coverage?

Reply:
I'll file a follow up :)